api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +565,134 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      if (crs != null) {
+        Preconditions.checkArgument(!crs.isEmpty(), "Invalid CRS: (empty string)");
+        Preconditions.checkArgument(
```

Comment:
i might have missed this, but why not just trim it?

Reply:
Constructing a GeometryType manually using `GeometryType.of(" crs_value  ")` and retrieve its CRS will result in a different value if we trim it. I've asked for it here https://github.com/apache/iceberg/pull/12346#discussion_r1996626058 and decided to throw exception to avoid any unexpected behavior for such cases. I can change it to silently trimming the input if you consider it a more appropriate approach.

Reply:
I don't think this check is needed. We don't do similar validations elsewhere. The responsibility of this class is to pass the incoming CRS without modification. There's no need to validate the string. Parsing should handle that.

Reply:
Removed trim. Trimming is handled by regex when parsing primitive type strings.